3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/registry/MeasurementProtocolAdapter.scala

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2012-2017 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics
+package snowplow.enrich.common
+package adapters
+package registry
+
+// Scalaz
+import scalaz._
+import Scalaz._
+
+// Iglu
+import iglu.client.{Resolver, SchemaKey}
+
+// This project
+import loaders.CollectorPayload
+
+/**
+ * Transforms a collector payload which conforms to a known version of the Google Analytics
+ * protocol into raw events.
+ */
+object MeasurementProtocolAdapter extends Adapter {
+
+  // for failure messages
+  private val vendorName = "MeasurementProtocol"
+  private val vendor = "com.google.analytics.measurement-protocol"
+  private val protocolVersion = "v1"
+  private val protocol = s"$vendor-$protocolVersion"
+  private val format = "jsonschema"
+  private val schemaVersion = "1-0-0"
+
+  case class HitTypeData(schemaUri: String, translationTable: Map[String, String])
+
+  val hitTypeData = Map(
+    "pageview" -> HitTypeData(
+      SchemaKey(vendor, "page_view", format, schemaVersion).toSchemaUri,
+      Map(
+        "dl" -> "documentLocationURL",
+        "dh" -> "documentHostName",
+        "dp" -> "documentPath",
+        "dt" -> "documentTitle"
+      )
+    ),
```

Comment:
I haven't found a better way to maintain this mapping yet if someone has any ideas they're welcome.